### PR TITLE
Add support for `Config` module, closes #2346

### LINKED ISSUE #2346: Add a Schema.config for decoding Configs using a Schema.
### What is the problem this feature would solve?

It would be convenient to be able to use a Schema instead one of the built in Config primitives for decoding a Config var.  

### What is the feature you are proposing to solve the problem?

Add a `Schema.config :: (Schema<A, string, R>, name?: string) -> Effect<Config<A>, never, R>` function.
### What alternatives have you considered?

None.

### PR DESCRIPTION
**Pull request by gcanti**
## Type

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Related

- Closes #2346

### CODE REVIEW

**Review by vecerek — APPROVED** [review]
Looks very similar to my suggestion in https://github.com/Effect-TS/effect/issues/2763#issuecomment-2116341818

Instant approve 😀👍

File: .changeset/eleven-pens-kick.md [hunk 1]
```diff
@@ -2,4 +2,4 @@
 "@effect/schema": patch
 ---
 
-Add support for Enums in Record, closes #2811
+Add support for `Enums` in `Record`, closes #2811
```
Comment (vecerek) [inline .changeset/eleven-pens-kick.md:5]: Changeset from a different branch?

**Comment by jessekelly881** [comment]
Looks great!

**Comment by gcanti** [comment]
> Looks very similar to my suggestion in https://github.com/Effect-TS/effect/issues/2763#issuecomment-2116341818

Yes 👍 , this issue lacked a proposal and your implementation is nice and works, except for an extra `name` in 
```diff
-InvalidData([name], TreeFormatter.formatErrorSync(error))
+InvalidData([], TreeFormatter.formatErrorSync(error))
```

**Pull request merged**